openhands/cli/main.py

```diff
@@ -288,7 +288,12 @@ async def main(loop: asyncio.AbstractEventLoop):
 
     # Use settings from settings store if available and override with command line arguments
     if settings:
-        config.default_agent = args.agent_cls if args.agent_cls else settings.agent
+        # Handle the agent class setting with proper type handling
```

Comment:
This one is not a lint change. Is that intentional?

Reply:
Fixed to make it more logical.